packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -2,55 +2,159 @@
     resendSecondaryEmailCode(input: $input) {
       clientMutationId
     }
   }
 `;
 
+export const MAKE_EMAIL_PRIMARY_MUTATION = gql`
+  mutation updatePrimaryEmail($input: EmailInput!) {
+    updatePrimaryEmail(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const DELETE_EMAIL_MUTATION = gql`
+  mutation deleteSecondaryEmail($input: EmailInput!) {
+    deleteSecondaryEmail(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
 type UnitRowSecondaryEmailContentAndActionsProps = {
   secondary: Email;
   isLastVerifiedSecondaryEmail: boolean;
 };
 
 export const UnitRowSecondaryEmail = () => {
   const account = useAccount();
+  const navigate = useNavigate();
+  const [verifiedAction, setVerifiedAction, verified] = useVerifiedAction();
   const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
   const [email, setEmail] = useState<string>();
-  const navigate = useNavigate();
+  const [alertMessage, setAlertMessage] = useState<[string, AlertBarType]>();
+
   const secondaryEmails = account.emails.filter((email) => !email.isPrimary);
   const hasAtLeastOneSecondaryEmail = !!secondaryEmails.length;
   const lastVerifiedSecondaryEmailIndex = secondaryEmails
     .map((email) => email.verified)
     .lastIndexOf(true);
 
-  // TODO: DRY this up and don't import `sentryMetrics` into every component
-  // needing a mutation - we must have an `onError` option in mutations
-  // to allow tests to pass but providing one prevents an error from actually
-  // throwing so we need to manually report it to Sentry.
-  // See https://github.com/apollographql/react-apollo/issues/2614
-  const [resendSecondaryEmailCode, { error }] = useMutation(
-    RESEND_SECONDARY_EMAIL_CODE_MUTATION,
-    {
-      onError: (error) => {
-        sentryMetrics.captureException(error);
-        revealAlertBar();
-      },
-      onCompleted: () => {
-        navigate('/beta/settings/emails/verify', { state: { email } });
-      },
-    }
+  const showSuccess = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'success']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const showError = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'error']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const [resendEmailCode] = useHandledMutation(RESEND_EMAIL_CODE_MUTATION, {
+    onCompleted() {
+      navigate('/beta/settings/emails/verify', { state: { email } });
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem re-sending the verification code.`);
+      throw error;
+    },
+  });
+
+  const [makeEmailPrimary] = useHandledMutation(MAKE_EMAIL_PRIMARY_MUTATION, {
+    onCompleted() {
+      showSuccess(`${email} is now your primary email.`);
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem changing your primary email.`);
+      throw error;
```

Comment:
I debated this -- useHandledMutation will capture this throw and send it to Sentry. The only reason I have left it here is because we currently don't know of any concrete reasons this would or would not throw. Because of that I think it would be best to keep this here and remove it down the road once Settings is more stable/released.

Reply:
Ah ok, that's fine by me.